test/MouseAdapterSpec.js

```diff
@@ -270,5 +270,123 @@ define(function(require) {
+                    expect(adapter._calculateDeltaDirection([1, 2, 1, 2, 1])).toBe(0);
+                    expect(adapter._calculateDeltaDirection([10, 9, 5, 9, 10])).toBe(0);
+                });
+
+                it('should begin at the specified offset when passed in', function() {
+                    expect(adapter._calculateDeltaDirection([1, 1, 1, 1, 2, 2, 2, 2], 0)).toBe(1);
+                    expect(adapter._calculateDeltaDirection([1, 1, 1, 1, 2, 2, 2, 2], 4)).toBe(-1);
+                });
+
+                it('should stop at the specified length when passed in', function() {
+                    expect(adapter._calculateDeltaDirection([1, 1, 1, 1, 2, 2, 2, 2], 0, 4)).toBe(0);
+                    expect(adapter._calculateDeltaDirection([1, 1, 1, 1, 2, 2, 2, 2], 4, 4)).toBe(0);
+                });
+
+            });
+
+            describe('_detectNegativeSpike', function() {
+                // Test arrays in this block should be of size MouseAdapter.DELTA_ARRAY_SIZE,
+                // representing mouse event deltas received in chronological order from index 0.
+
+                it('should detect sufficiently steep negative spikes in data', function() {
+                    expect(adapter._detectNegativeSpike([63.5, 62, 59, 110, 49.5, 45, 41, 37.5, 34.5, 0.5])).toBe(true);
+                    expect(adapter._detectNegativeSpike([29, 26.5, 24, 21.5, 20, 18, 32, 14, 13, 0.5])).toBe(true);
+                    expect(adapter._detectNegativeSpike([6, 5.5, 5, 5, 4.5, 4, 3.5, 3, 3, 0.5])).toBe(true);
+                });
+
+                it('should not react to noisy spikes in data', function() {
+                    expect(adapter._detectNegativeSpike([30, 63.5, 65.5, 136, 66.5, 63.5, 62, 59, 110, 49.5])).toBe(false);
+                    expect(adapter._detectNegativeSpike([6, 5.5, 5, 5, 4.5, 4, 3.5, 3, 10, 2])).toBe(false);
+                });
+            });
+
+            describe('_detectDeltaIncrease', function() {
+                // Test arrays in this block should be of size MouseAdapter.DELTA_ARRAY_SIZE,
+                // representing mouse event deltas received in chronological order from index 0.
+
+                it('should detect increasing trends in data that are preceded by a decreasing trend', function() {
+                    expect(adapter._detectDeltaIncrease([9, 22, 6, 6, 5.5, 5, 4.5, 4, 6.5, 10.33333333])).toBe(true);
+                    expect(adapter._detectDeltaIncrease([2.5, 2.5, 2, 2, 2, 1.5, 0.5, 2.5, 3, 3.5])).toBe(true);
+                });
+
+                it('should not react to increasing trends in data that are NOT preceded by a decreasing trend', function() {
+                    expect(adapter._detectDeltaIncrease([8, 9, 16, 17.5, 19, 19, 25, 54, 56.5, 58.5])).toBe(false);
+                    expect(adapter._detectDeltaIncrease([19, 26, 26.5, 29.5, 32, 34, 26.5, 40.5, 19, 72.5])).toBe(false);
+                });
```

Comment:
I could use ints, but the data wouldnt be as authentic.  I can add comments which describe the trends.

Given no user interaction, momentum only decreases over time, it will never increase on its own.  `_detectDeltaIncrease` works by looking for increasing trends in the data, preceded by decreasing trends, which could only mean that the user scrolled to add more momentum.

The function samples over sets of 5 data points (just enough to smooth out noise, but not too much to make it slow), requiring 10 points to look for trends.  In the first example, the data (which represents the entire recent scroll history) can be partitioned into two blocks of 5 `[9, 22, 6, 6, 5.5] [5, 4.5, 4, 6.5, 10.33333333]` and the function checks to see that block 1 is decreasing, and block two is increasing, indicating the user scrolled.